storage/src/mmr/verification.rs

```diff
@@ -2,9 +2,13 @@ use crate::mmr::hasher::Hasher;
 use crate::mmr::iterator::PeakIterator;
 use commonware_cryptography::{Digest, Hasher as CHasher};
 
+/// A `Proof` contains the information necessary for proving the inclusion of an element, or some
+/// range of elements, in the MMR from its root hash. The `hashes` vector contains: (1) the peak
+/// hashes other than those belonging to trees containing some elements within the range being
+/// proven, followed by: (2) the nodes in the remaining perfect trees necessary for reconstructing
+/// their peak hashes from the elements within the range. Both segments are ordered by decreasing
+/// height.
 #[derive(Clone, Debug, PartialEq, Eq)]
-/// A Proof contains the information necessary for proving the inclusion of an element, or some
-/// range of elements, in the MMR.
 pub struct Proof {
```

Comment:
We'll need to add `serialize/deserialize` functions for this. We don't have to do it here but expect it to follow the pattern used in `cryptography`/`consensus` (we manually pack to ensure encoding is always deterministic and minimal).

Reply:
ok, will have to learn a bit about how that works.

Reply:
These are good minimal examples:
* https://github.com/commonwarexyz/monorepo/blob/01180c29e5771d91fca0455ef2bb98542befad2e/cryptography/src/bls12381/primitives/group.rs#L195-L210
* https://github.com/commonwarexyz/monorepo/blob/01180c29e5771d91fca0455ef2bb98542befad2e/consensus/src/simplex/prover.rs#L41-L97